src/controls/mod.rs

```diff
@@ -369,13 +369,18 @@ impl Program for ControlState {
     }
 }
 
-impl ControlState {
-    fn layout_row(row: Row<'_, Message, Renderer>) -> Row<'_, Message, Renderer> {
-        row
-            .padding(5)
-            .spacing(10)
-            .width(Length::Fill)
-            .align_items(Alignment::Center)
-    }
+trait Component {
+    type LocalMessage;
```

Comment:
On second thought, I might prefer to call this associated type `Message` and the top-level message something like `GUIMessage`. `LocalMessage` is a long name for a type param